src/Consolonia.Core/Infrastructure/ArrowsAndKeyboardNavigationHandler.cs

```diff
@@ -124,8 +125,26 @@ private void OnKeyDown(object sender, KeyEventArgs e)
         {
             if (e.Handled) return;
 
+            if (e.Source is MenuItem) return;
```

Comment:
I'm wondering how does Avalonia work without this? May be we should drop a ticket to Avalonia same time?

Reply:
I don't really understand why we have the ArrowsAndKeyboardNavigatorHandler at all. I was thinking about this and probably the cleaner thing to do would be to register the OnKeyDown hook if the item is a MenuItem. 

I can give that a try if you think it's cleaner...

Reply:
> All I know is that if there is a MenuItem then the left/right/esc keys navigation keys logic messes up the keyboard navigation when in a menu system.

ok. It looks like a bug of Avalonia. MenuItem should set e.Handled if it handles arrow navigation. I will investigate after.